server/handlers.go

```diff
@@ -0,0 +1,135 @@
+package server
+
+import (
+	"crypto/md5"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/ASeegull/SmartFridge/server/database"
+	"github.com/gorilla/websocket"
+	log "github.com/sirupsen/logrus"
+)
+
+func getHash(stringToHash string) string {
+	hash := md5.Sum([]byte(stringToHash))
+	return hex.EncodeToString(hash[:])
+}
+
+var upgrader websocket.Upgrader
+
+func setUpgrader() {
+	upgrader = websocket.Upgrader{
+		ReadBufferSize:  serverConfig.ReadBufferSize,
+		WriteBufferSize: serverConfig.WriteBufferSize,
+	}
+}
+
+func agentAuthentication(w http.ResponseWriter, r *http.Request) {
+	tokenExample := getHash("userID+agentID+productName")
+	err := json.NewEncoder(w).Encode(tokenExample)
+
+	if err != nil {
+		log.Print(err)
+	}
+}
+
+func createWS(w http.ResponseWriter, r *http.Request) {
+	conn, err := upgrader.Upgrade(w, r, nil)
+
+	if err != nil {
+		log.WithField("error: ", err).Info("Problem with creation websocket")
+		return
+	}
+
+	fmt.Printf("New websocket connect with %s\n", r.Host)
+	wg.Add(1)
+	go wsListener(conn)
+}
+
+func wsListener(conn *websocket.Conn) {
+	defer wg.Done()
+	defer conn.Close()
+
+	for {
+		time.Sleep(time.Second * serverConfig.WebsocketSleep)
```

Comment:
Bad approach =)
`conn.ReadMessage()` and `conn.ReadJSON()` are already blocking